src/NServiceBus.RabbitMQ.Tests/RabbitMqContext.cs

```diff
@@ -99,30 +69,39 @@ public void TearDown()
             channelProvider?.Dispose();
         }
 
-        protected IncomingMessage WaitForMessage()
+        protected bool TryWaitForMessageReceipt()
         {
-            var waitTime = TimeSpan.FromSeconds(1);
+            IncomingMessage message;
+            return TryReceiveMessage(out message, incomingMessageTimeout);
+        }
 
-            if (Debugger.IsAttached)
+        protected IncomingMessage ReceiveMessage()
+        {
+            IncomingMessage message;
+            if (!TryReceiveMessage(out message, incomingMessageTimeout))
             {
-                waitTime = TimeSpan.FromMinutes(10);
+                throw new TimeoutException($"The message did not arrive within {incomingMessageTimeout.TotalSeconds} seconds.");
             }
 
-            IncomingMessage message;
-            receivedMessages.TryTake(out message, waitTime);
-
             return message;
         }
 
+        private bool TryReceiveMessage(out IncomingMessage message, TimeSpan timeout) =>
+            receivedMessages.TryTake(out message, incomingMessageTimeout);
```

Comment:
Oops, yes, will fix. The reason that I wanted the timeout to be passed into this method, is that `ReceiveMessage()` throws an exception which mentions the timeout period, and it felt a little ugly spreading the knowledge of that field across caller and callee.